### ISSUE TITLE
Unnecessary code gets bundled when adding a component

### ISSUE THREAD

**Issue by kubk** [XeleneStudio/TelegramUI#3]
Hi. The library has a lot of working components, which is great because users can already start integrating them into their projects. The issue I noticed is that even importing one component results in a lot of unnecessary bundled code, causing the bundle to bloat.

Here is my code example:

```tsx
<AppRoot>
  <SegmentedControl>
    {labels.map(({ value, label }) => (
      <SegmentedControl.Item
        key={value}
        selected={selected === value}
        onClick={() => setSelected(value)}
      >
        {label}
      </SegmentedControl.Item>
    ))}
  </SegmentedControl>
</AppRoot>;
```

It produces the following report generated by the `vite-bundle-visualizer`:

<img width="1256" alt="Screenshot 2024-03-11 at 09 47 14" src="https://github.com/XeleneStudio/TelegramUI/assets/22447849/db68bb9b-ebaa-4798-a7bb-a726742e1608">

Here we see a lot of tgui components (all the components I guess), Radix UI, a library to remove scroll. Reproduction repo: https://github.com/kubk/tgui-bundle-size-issue

**Why tree-shaking is important**
Users may be reluctant to use the library because importing even one component results in loading a lot of unnecessary code.

**Comment by mainsmirnov** [comment]
It seems like your Vite configuration might not be set up correctly. Our library supports tree shaking, and its bundle size is quite minimal. You can verify its size on BundlePhobia.

Additionally, you can test tree shaking with an appropriate build tool, such as `react-scripts`, which you can find here: [react-scripts on npm](https://www.npmjs.com/package/react-scripts).

Alternatively, consider using our template available at [XeleneStudio TGUI-Example on GitHub](https://github.com/XeleneStudio/TGUI-Example)

**Comment by kubk** [comment]
@mainsmirnov It's default Vite configuration. You can recreate it via `npm create vite@latest my-react-app -- --template react-ts` and then add tgui. 

`vite` has 11.7 millions downloads weekly: https://www.npmjs.com/package/vite
`react-scripts` has 3.2 millions: https://www.npmjs.com/package/react-scripts
It's even doubtful whether react-scripts is still maintained: https://github.com/facebook/create-react-app/issues/13503

Vite works fine with code splitting out of the box, the code splitting works for my telegram mini app and other dependencies, except tgui. I'd consider this worth looking from the tgui side, unless the library doesn't want to miss plethora of `vite` users.

**Comment by kubk** [comment]
Not sure if tgui is going to be included in the Telegram Mini App template that is being built by @heyqbnk, but that template is also based on Vite: https://github.com/Telegram-Mini-Apps/reactjs-template.

**Comment by heyqbnk** [comment]
Hey! Yeah, I have plans on adding this package to the templates based on React. I tried to use it several days ago, but met some problems, and decided to postpone the idea of integrating it. Moreover, it seems like the package is hardly bound to the Telegram SDK and I am not sure if it can be used along with @tma.js.

Thanks for your mention, because I forgot to check the bundle size with this package included. React JS template already uses **@tonconnect/ui-react** (~300kb) and **react-router-dom** (~200kb). Adding 1 more huge library will just blow my mind. So, we are not going not add this package until its size is optimal and problems are solved.

Waiting for grant to be completed and coming back to see the results. I am currently developing the separate package working with Telegram SDK and @tma.js using Solid.

**Comment by mainsmirnov** [comment]
We took the build settings from this library. Could you check if it works there? If the problem remains relevant, then I’m afraid that the problem is still in your config

**Comment by heyqbnk** [comment]
Due to the reason, that bundlephobia is currrently down, I have tried [bundlejs.com](https://bundlejs.com/?q=%40xelene%2Ftgui&treeshake=%5B%7B+Button+%7D%5D) to check the bundle size. 

![image](https://github.com/XeleneStudio/TelegramUI/assets/34907325/c552e579-2371-4c8b-88a5-1568f4cb9d8e)

Not sure that config could be the problem, because we use default config that already includes tree shaking and works fine with other package

**Comment by mainsmirnov** [comment]
We adopted the build configurations from the [library](https://github.com/VKCOM/VKUI). If the issue also exists there, it might be necessary to adjust your setup.

My attempt to replicate the issue with webpack was unsuccessful. Checking the library's bundle reveals that files are individually compiled, ensuring full tree-shaking support.

We will take a closer look look at it

**Comment by heyqbnk** [comment]
I have tried this code locally:

```ts
import { Typography } from '@xelene/tgui';

console.log(Typography);
```

After building the code using Vite (which uses Rollup, minifying the code via `terser`), I receive 204kb bundle size. In case, I import only React, the bundle size becomes 142kb.

I have tried this code to see if we pull whole package importing the only `Typography` component:

```ts
import * as tgui from '@xelene/tgui';

console.log(tgui);
```

Bundle size is 295kb. So, we don't pull the whole package importing the only one component, but **half** of it. 

Then, I tried to import this component using this way:

```ts
import { Typography } from '@xelene/tgui/dist/components/Typography';

console.log(Typography);
```

Bundle size is now 10kb.

So, I assume, that package contains side-effects which **can't be tree-shaken**. Importing any component from the root will lead to importing all side-effects. That's why importing `Typography` not from the root costs way less bundle size.

Let me know if I am missing something.

**Comment by mainsmirnov** [comment]
Thank you for providing the detailed repository with a demonstration. We have resolved the issue in version 2.0.4

<img width="1723" alt="Screenshot 2024-03-13 at 12 52 30 PM" src="https://github.com/XeleneStudio/TelegramUI/assets/45998835/adca322c-e111-466c-9857-9535438e746f">

**Comment by kubk** [comment]
@mainsmirnov Thank you! Can confirm it fixed the issue 👍
